flask/boxwise_flask/models/user.py

```diff
@@ -1,33 +1,62 @@
-from peewee import CharField, DateField, DateTimeField
+from boxwise_flask.db import db
+from boxwise_flask.models.language import Language
+from boxwise_flask.models.usergroup import Usergroup
+from boxwise_flask.models.usergroup_base_access import UsergroupBaseAccess
+from peewee import (
+    SQL,
+    CharField,
+    DateField,
+    DateTimeField,
+    ForeignKeyField,
+    IntegerField,
+)
 from playhouse.shortcuts import model_to_dict
 
-from ..db import db
-from .usergroup_base_access import UsergroupBaseAccess
-
 
 class User(db.Model):
-    name = CharField(column_name="naam")
-    email = CharField()
-    usergroup_id = CharField(column_name="cms_usergroups_id")
-    valid_firstday = DateField()
-    valid_lastday = DateField()
-    lastlogin = DateTimeField()
+    usergroup = ForeignKeyField(
+        column_name="cms_usergroups_id", field="id", model=Usergroup, null=True
+    )
+    created = DateTimeField(null=True)
+    created_by = ForeignKeyField(
+        column_name="created_by", field="id", model="self", null=True
+    )
+    deleted = DateTimeField(null=True, default=None)
+    email = CharField(null=True, unique=True)
+    is_admin = IntegerField(constraints=[SQL("DEFAULT 0")])
+    language = ForeignKeyField(
+        column_name="language", field="id", model=Language, null=True
+    )
     lastaction = DateTimeField()
+    lastlogin = DateTimeField()
+    modified = DateTimeField(null=True)
+    modified_by = ForeignKeyField(
+        column_name="modified_by", field="id", model="self", null=True,
+    )
+    name = CharField(column_name="naam", constraints=[SQL("DEFAULT ''")])
+    pass_ = CharField(column_name="pass", constraints=[SQL("DEFAULT ''")])
+    resetpassword = CharField(null=True)
+    valid_firstday = DateField(null=True)
+    valid_lastday = DateField(null=True)
 
     class Meta:
         table_name = "cms_users"
 
     def __str__(self):
-        return self.name, self.organisation_id
+        return self.name
 
     @staticmethod
     def get_all_users():
-        return User.select().order_by(User.name)
+        return list(User.select().order_by(User.name))
 
     @staticmethod
-    def get_user(email):
+    def get_from_email(email):
         user = User.select().where(User.email == email).get()
-        base_ids = UsergroupBaseAccess.get_all_base_id_for(user.usergroup_id)
+        base_ids = []
+        if user.usergroup:
+            base_ids = UsergroupBaseAccess.get_all_base_id_for_usergroup_id(
+                user.usergroup.id
+            )
         # base_ids is a peewee ModelSelect (so, many objects).
         # convert to dict 1 at a time,
         # and pull the base_id from that dict, and put in a list
```

Comment:
I don't exactly understand how this works... the `User` model does not even have a `base_id`?

Reply:
So this is a little confusing and I didn't write this originally :confused: . But because the user graphql query has a base id as part of the type def its appended to the object that returns in the case it exists. I shall move to the parent function call to make it more obvious how this works. it should also probably be camp_id. which will involved a frontend change as well :+1:

Reply:
Ok nevermind :) maybe something for a tiny refactoring ticket rather @aerinsol

Reply:
That's weird. The user query has a base option? Not sure I understand how that happened, let's discuss on Slack